Sources/Extensions/CALayer+Animation.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  CALayer+Animation.swift
+//  MessageStackView
+//
+//  Created by Ben Shutt on 27/06/2020.
+//  Copyright © 2020 3 SIDED CUBE APP PRODUCTIONS LTD. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+// MARK: - CALayer
+
+extension CALayer {
+    
+    /// Animate transform from `from` to `to`
+    /// - Parameters:
+    ///   - duration: Animation duration
+    ///   - from: Start state
+    ///   - to: End state
+    func transformAnimation(
+        duration: TimeInterval,
+        from: [CGFloat] = [0, 0, 0],
+        to: [CGFloat] = [1, 1, 1]
+    ) -> CABasicAnimation {
+        
+        guard from.count == 3 else { fatalError("Invalid from argument") }
```

Comment:
I'm not sure if `fatalError` is the best way to go here, could we not provide a `struct` property of maybe like:
```
struct Scale {
    let x: CGFloat,
    let y: CGFloat,
    let z: CGLoat
}
```

Reply:
@simonmitchell Yeah totally agree, using a `Vector3` or something is much nicer, I believe this was taken from Apple docs but I will update 😄